src/data/selectors/grades.js

```diff
@@ -95,9 +95,25 @@ const formatMinAssignmentGrade = (state, assignmentId, percentGrade) => {
   return percentGrade;
 };
 
+const formatMaxCourseGrade = (state, percentGrade) => {
```

Comment:
I think we have a functional programming pattern we can extract here: these functions have a common shape which can be its own function. I think we could represent these different iterations of the same idea using function composition as our means of combination, to DRY up this code between how assignment & course grades are handled.

Feel free to bug me if I don't follow up with code on this comment; don't want the description of what I'm after to stay this abstract

Reply:
Also these don't actually need `state`

Reply:
```
const composeFilters = (...predicates) => (percentGrade, options = {}) =>
  predicates.reduce((accum, predicate) => {if(predicate(percentGrade, options)) {return null;} return accum;});
```
used e.g.
```
formatMaxCourseGrade = composeFilters(percentGradeIsMax);
formatMinAssignmentGrade = composeFilters(
  percentGradeIsMin,
  (percentGrade, { assignmentId }) => !assignmentId,
);
```